theano/tensor/signal/pool.py

```diff
@@ -84,156 +85,181 @@ def pool_2d(input, ds, ignore_border=None, st=None, padding=(0, 0),
+            " On the GPU, using ignore_border=True is needed to use cuDNN."
+            " When using ignore_border=False and not using cuDNN, the only"
+            " GPU combination supported is when"
+            " `ds == st and padding == (0, 0, 0) and mode == 'max'`."
+            " Otherwise, the convolution will be executed on CPU.",
+            stacklevel=2)
+        ignore_border = False
+    op = Pool(ignore_border, ndim=3, mode=mode)
+    output = op(input, ds, st, padding)
+    return output
 
 
 class Pool(OpenMPOp):
     """
-    For N-dimensional tensors, consider that the last two dimensions span
-    images. This Op downsamples these images by taking the max, sum or average
-    over different patch.
-
-    The constructor takes the max, sum or average or different input patches.
+    This Op downsamples the last N dimensions of the input by taking the max,
+    sum or average over different patches.
 
     Parameters
     ----------
-    ds : list or tuple of two ints
-        Downsample factor over rows and column.
-        ds indicates the pool region size.
+    ds : list or tuple of N ints
+        Downsample factor over rows, columns etc.
+        ds indicates the size of the pooling region.
     ignore_border : bool
-        If ds doesn't divide imgshape, do we include an extra row/col
+        If ds doesn't divide imgshape, do we include an extra row/col/slice
         of partial downsampling (False) or ignore it (True).
-    st : list or tuple of two ints or None
-        Stride size, which is the number of shifts over rows/cols to get the
+    st : list or tuple of N ints or None
+        Stride size, which is the number of shifts over rows/cols/slices to get the
         next pool region. If st is None, it is considered equal to ds
         (no overlap on pooling regions).
-    padding: tuple of two ints
-        (pad_h, pad_w), pad zeros to extend beyond four borders of the images,
-        pad_h is the size of the top and bottom margins, and pad_w is the size
-        of the left and right margins.
+    padding : tuple of N ints or None
+        For each downsampling dimension, this specifies the number of zeros to
+        add as padding on both sides. For 2D and (pad_h, pad_w), pad_h specifies the
+        size of the top and bottom margins, pad_w specifies the size of the left and
+        right margins. No padding is added if padding is None.
     mode : {'max', 'sum', 'average_inc_pad', 'average_exc_pad'}
         ('average_inc_pad' excludes the padding from the count,
         'average_exc_pad' include it)
+    ndim : int
+        The number of pooling dimensions N.
+        If this number is not specified, the default is set to the
+        (input.ndim - 2), assuming that the first two dimensions of the input
+        are non-pooling dimensions.
 
     """
 
-    __props__ = ('ignore_border', 'mode')
+    __props__ = ('ndim', 'ignore_border', 'mode')
```

Comment:
Try to keep the same order as in the \__init__, so add it at the end of the list.